### DISCUSSION TITLE
UWP Port possibility

### DISCUSSION THREAD

**Discussion by marcelofrau** [joncampbell123/dosbox-x#4186] [Ideas]
Hello, 

I am seeing a lot of traction in UWP applications being developed for emulation and general purpose apps to run in the Xbox environment, specially with the possibility to everyone put their console in Dev Mode, and install custom and public available UWP applications, would be a very nice opportunity to have the fantastic dosbox-x running in a modern console. 

The same was done for other emulators, and there is even a retroarch core for dosbox SVN running in the Xbox consoles, but a standalone and much better version of dosbox would be nice to be available there. 

I don't have much UWP knowledge, but I am a software engineer for a long time already, not so much experienced with C++ but I'll try to see if I could help to achieve this somehow. 

I've seen some other ports using SDL already in UWP, so I think it should not be a problem to port dosbox-x, but this is just speculation from my side.

Thank you,
Best regards

**Comment by joncampbell123** [comment]
SDL2 itself seems to support UWP. If you want to fork DOSBox-X to make a UWP version, that's perfectly fine with me. Perhaps then you could merge your changes back with \#ifdef's to separate WIN32 from UWP code and it could be one of the build targets in the VS2019 project/solution files. Does that sound good?

**Reply by marcelofrau** [reply]
it sounds awesome.. let me see if I can work on this.. will pull the last changes from dosbox and build again on the VS2019 once I'm home.. and will start learning how to work with UWP and see how I would do the port.. 

thanks for the tips.

**Comment by joncampbell123** [comment]
DOSBox-X uses C++ classes, some templates, and a few STL libraries, but overall keeps the code at a mixed C and C++ level. Pointers are used heavily in this code. Contrary to the "wisdom" in the industry, pointers are not evil, they're fairly simple to manage so long as you don't do anything exotic with them or lose track of memory. Some code, like the scalers, are fairly heavy in #define macro use.

I don't expect SDL1 to make it as UWP code, especially the version used in DOSBox-X that is heavily modified from the original library.

**Reply by marcelofrau** [reply]
ok, so let me check more about the SDL2 build and see what is the effort to port this.

**Reply by marcelofrau** [reply]
I mean, I know pointers are not evil, I am just not so used to deal with them, but I think this would be the least of my problems to be honest (but well, you never know), I mean, I worked with some JNI libraries wrappers and had to know some C and C++, and also worked with QT and C++, it was some time ago, but I can consider myself a little rusted. let's see..

**Comment by aybe** [comment]
@marcelofrau There's no need to post a question twice, continue the chat where you originally [started it](https://github.com/joncampbell123/dosbox-x/issues/4187), in the issues tab.
